src/data/nat/factorization.lean

```diff
@@ -42,7 +43,30 @@ namespace nat
 
 /-- `n.factorization` is the finitely supported function `ℕ →₀ ℕ`
  mapping each prime factor of `n` to its multiplicity in `n`. -/
-noncomputable def factorization (n : ℕ) : ℕ →₀ ℕ := (n.factors : multiset ℕ).to_finsupp
+def factorization (n : ℕ) : ℕ →₀ ℕ :=
+{ support := n.factors.to_finset,
+  to_fun := λ p, if p.prime then padic_val_nat p n else 0,
+  mem_support_to_fun :=
+    begin
+      intro a,
+      rw list.mem_to_finset,
+      by_cases hn0 : n = 0,
+      { sorry, },
+      rw mem_factors hn0,
+      simp only [ne.def, ite_eq_right_iff, exists_prop],
+      sorry,
+    end }
```

Comment:
Using the lemma `dvd_iff_padic_val_nat_ne_zero` that I suggested in another comment:
````lean
mem_support_to_fun :=
    begin
      intro a,
      rw list.mem_to_finset,
      rcases eq_or_ne n 0 with rfl | hn0, { simp },
      rw mem_factors hn0,
      simp only [ne.def, ite_eq_right_iff, not_forall, exists_prop, and.congr_right_iff],
      intro pa,
      haveI : fact (prime a) := fact_iff.mpr pa,
      exact dvd_iff_padic_val_nat_ne_zero hn0,
    end }
````

Reply:
added